### ISSUE TITLE
Visualise test coverage in BPMN models

### ISSUE THREAD

**Issue by remcowesterhoud** [camunda-cloud/camunda-cloud-testing#23]
**Description**

We want to visualise the test coverage of a process. This allows us to gain insight in how well our processes are tested, and which part of the process requires some more attention to be fully covered.

**Relevant links**
https://camunda.com/blog/2020/12/measure-your-coverage/

**Comment by berndruecker** [comment]
@falko :-)

**Comment by falko** [comment]
It would be great, if we build it language-independent this time, i.e. not tied to Java and just using public APIs. The general approach and report templates from the [Camunda Platform 7 Process Test Coverage](https://github.com/camunda-community-hub/camunda-process-test-coverage) should be easily reusable. One basically queries the history audit trail and replaces the placeholder `{{__REPORT_JSON_PLACEHOLDER__}}` in the [report template](https://github.com/camunda-community-hub/camunda-process-test-coverage/blob/master/extension/report-generator/src/main/resources/bpmn.report-template.html) with the data.

**Comment by ingorichtsmeier** [comment]
I think the people maintaining the process test coverage extension are currently working on this: [add examples for platform 8](https://github.com/camunda-community-hub/camunda-process-test-coverage/commit/f77805d8aa153d130ae471b394661823e1677829)

**Comment by rohwerj** [comment]
Just for reference the community extension https://github.com/camunda-community-hub/camunda-process-test-coverage does support Camunda8 as @ingorichtsmeier mentioned above.
It can be integrated with JUnit5 and Spring Testing.

**Comment by saig0** [comment]
> [!Note]
> Heads up! We are building a new Java testing library for Camunda 8.6. The new library will replace Zeebe Process Test.
> Read more about upcoming changes [here](https://github.com/camunda/issues/issues/751) and stay tuned for updates. :rocket:

**Comment by rohwerj** [comment]
@saig0 Do you already have some ideas, how exactly this will look? Will there still be a way to access the record stream in the tests? I would like to adapt https://github.com/camunda-community-hub/camunda-process-test-coverage as soon as there's some code to test it with.

**Comment by saig0** [comment]
@rohwerj Yes, I have some ideas. :bulb: The new testing library will run the complete Camunda Docker Image via Testcontainers instead of running an in-memory Zeebe workflow engine. We will share more details later when we have some code ready.

> Will there still be a way to access the record stream in the tests?

No. Since the Camunda engine runs inside Docker, you must use the Camunda REST API to access the data.

---

We plan to include a process coverage feature in the new testing library. It will be inspired by the community project. At least in the beginning, it will support fewer features.

**Comment by berndruecker** [comment]
@saig0 I was surprised to see we want to remove the in-memory testing capability. Is there a related issue I can check the reasoning (or discuss into the decision maybe)?

**Comment by saig0** [comment]
:fast_forward: Please see the issue https://github.com/camunda/issues/issues/751 for more details about the new testing library. We can discuss 
any questions and concerns about the solution in the comments.

**Comment by saig0** [comment]
Closing as out of scope for maintenance mode. We will not work on features or non-critical changes.
